.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,51 @@
+name: "Code scanning"
+
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: '0 2 * * 4'
+
+jobs:
+  CodeQL-Build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
+        # We must fetch at least the immediate parents so that if this is
+        # a pull request then we can checkout the head.
+        fetch-depth: 2
+
+    # If this run was triggered by a pull request event, then checkout
+    # the head of the pull request instead of the merge commit.
+    - run: git checkout HEAD^2
+      if: ${{ github.event_name == 'pull_request' }}
+      
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
+      # Override language selection by uncommenting this and choosing your languages
+      # with:
+      #   languages: go, javascript, csharp, python, cpp, java
+
+    # Autobuild attempts to build any compiled languages  (C/C++, C#, or Java).
+    # If this step fails, then you should remove it and run the build manually (see below)
+    - name: Autobuild
+      uses: github/codeql-action/autobuild@v1
```

Comment:
This is only for C/C++, C# and Java. Should this maybe be commented, while adding an issue to enable building the project in another way?

Reply:
The action detects that the project is Go and reports that "None of the languages in this project require extra build steps" so I think it's fine like this. I think it's best to leave the action in there, just in case GitHub decides to update the action in the future.